charts/karmada/values.yaml

```diff
@@ -296,6 +296,7 @@ controllerManager:
   featureGates:
     ## @param PropagateDeps is a feature gate for the controllerManager to allow propagate dependent respurce to workloads.
     PropagateDeps: false
+  controllers: ""
```

Comment:
Use a array to define the controllers feild?  like: `controllers: [ "a", "b" ]`

Reply:
But for the value of `--controllers` flag, it's a string.

Reply:
We can spport a template to parse the array as a string. I don't know whether it is necessary to do it

Reply:
For the users, there is no significant improvement in using arrays. I think it's fine to keep the status quo.

Reply:
@Poor12 Can you share the test result here? I want to know how to use this flag.

and @calvin0327 can you give an example for the array way? Let's see which one is better.

Reply:
For example:
If you want to disable hpa controller:
```
featureGates:
    ## @param PropagateDeps is a feature gate for the controllerManager to allow propagate dependent respurce to workloads.
    PropagateDeps: false
  controllers: "*,-hpa"
```

Then use `helm template` to render yamls:
```
 command:
            - /bin/karmada-controller-manager
            - --kubeconfig=/etc/kubeconfig
            - --bind-address=0.0.0.0
            - --cluster-status-update-frequency=10s
            - --secure-port=10357
            - --leader-elect-resource-namespace=default
            - --v=2
            - --controllers=*,-hpa
            - --feature-gates=PropagateDeps=false
```

Reply:
value.yaml:
```yaml
controllers: [ "a", "b", "-c", "*" ]
```

template:
```yaml
{{- define "karmada.controllerManager.controllers" -}}
  {{- if (not (empty .controllers)) }}
    {{- $controllersFlag := "" -}}
    {{- range .controllers }}
      {{- if not (empty (toString .)) }}
        {{- $controllersFlag = append $controllersFlag . -}}
      {{- end }}
    {{- end }}

    {{- if gt (len $controllers) 0 }}
      {{- $controllersFlag := trimSuffix "," $controllersFlag  | nospace -}}
      {{- printf "%s=%s" "--controllers" $controllersFlag -}}
    {{- end }}
  {{- end }}
{{- end -}}
```

uasage:
```yaml
- --kubeconfig=/etc/kubeconfig
- --bind-address=0.0.0.0
{{- if .Values.controllers }}
- {{ include "karmada.controllerManager.controllers" (dict "controllers" .Values.controllers) }}
{{- end }}
```

@Poor12 @RainbowMango That's the probable idea, How do you think?

Reply:
I‘d prefer to string one instead of the slice. Because it is more straightforward that we don't need to involve much conversion work.

Reply:
indeed, The way is complicated

Reply:
Thanks. I guess we can move forward now.